### PR TITLE
[12.0][MIG] account_invoice_check_identical_invoice

### PR DESCRIPTION
**Pull request by vvrossem**
#### [Task](https://gestion.coopiteasy.be/web#id=4623&view_type=form&model=project.task&menu_id=338&action=479)

changes : 
- since in Odoo 12.0, the Date and Datetime fields are now handled in the ORM as date objects, no need to convert `dt.date.today()` into a `str`

### CODE REVIEW

File: account_invoice_check_identical_invoice/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+{
+    "name": "Account Invoice Check Identical Invoice",
+    "version": "12.0.1.0.0",
+    "depends": ["base", "account",],
+    "author": "Coop IT Easy SCRLfs",
+    "category": "Accounting & Finance",
+    "website": "https://coopiteasy.be",
+    "description": """
+        This module requires to check this box to validate the invoice 
+        if invoices with the same partner, invoice date and totam alount already
```
Comment (robinkeunen) [inline account_invoice_check_identical_invoice/__manifest__.py]: Move description to DESCRIPTION.rst and add key summary in the __manifest__. You can install pre-commit for theses checks.

Also here, `totam alount` should be `total amount`

File: account_invoice_check_identical_invoice/models/account_invoice.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+    @api.onchange("partner_id", "date_invoice")
+    @api.multi
+    def _onchange_check_identical_invoice(self):
+        self._check_identical_invoice()
+
+    @api.multi
+    def _check_identical_invoice(self):
+        for invoice in self:
+            Invoice = self.env["account.invoice"]
```
Comment (robinkeunen) [inline account_invoice_check_identical_invoice/models/account_invoice.py]: You could put it out of the loop.

File: account_invoice_check_identical_invoice/models/account_invoice.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            def invoiced_today(i):
+                return (
+                    i.date_invoice == dt.date.today()
+                    and not invoice.date_invoice
+                )
+
+            duplicate_invoices = partner_invoices.filtered(
+                lambda i: equal_amount(i)
+                and (same_date(i) or invoiced_today(i))
+            )
```
Comment (robinkeunen) [inline account_invoice_check_identical_invoice/models/account_invoice.py:57]: good ol' Robin, did you notice the [closure](https://en.wikipedia.org/wiki/Closure_(computer_programming))?

File: account_invoice_check_identical_invoice/models/account_invoice.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+            if duplicate_invoices:
+                invoice.identical_invoice_detected = True
+            else:
+                invoice.identical_invoice_detected = False
```
Comment (robinkeunen) [inline account_invoice_check_identical_invoice/models/account_invoice.py]: Could be
```py
invoice.identical_invoice_detected = bool(duplicate_invoices)
```

**Review by robinkeunen — CHANGES_REQUESTED** [review]
Small change needed to the __manifest__ and some suggestions. Otherwise 👌

File: account_invoice_check_identical_invoice/__manifest__.py (hunk 1 above)
Comment (vvrossem) [reply account_invoice_check_identical_invoice/__manifest__.py]: done

File: account_invoice_check_identical_invoice/models/account_invoice.py (hunk 2 above)
Comment (vvrossem) [reply account_invoice_check_identical_invoice/models/account_invoice.py]: done

File: account_invoice_check_identical_invoice/models/account_invoice.py (hunk 3 above)
Comment (vvrossem) [reply account_invoice_check_identical_invoice/models/account_invoice.py:57]: sexy!

File: account_invoice_check_identical_invoice/models/account_invoice.py (hunk 4 above)
Comment (vvrossem) [reply account_invoice_check_identical_invoice/models/account_invoice.py]: done

File: account_invoice_check_identical_invoice/__manifest__.py (hunk 1 above)
Comment (robinkeunen) [reply account_invoice_check_identical_invoice/__manifest__.py]: sorry, that was misleading. I'll setup pre-commit here next time my compulsive disorders hit.

**Review by robinkeunen — APPROVED** [review]

**Pull request merged**